sound/soc/soc-topology.c

```diff
@@ -1156,18 +1156,22 @@ static int soc_tplg_dapm_graph_elems_load(struct soc_tplg *tplg,
 			SNDRV_CTL_ELEM_ID_NAME_MAXLEN)
 			return -EINVAL;
 
-		route.source = elem->source;
-		route.sink = elem->sink;
-		route.connected = NULL; /* set to NULL atm for tplg users */
+		route = kzalloc(sizeof(*route), GFP_KERNEL);
+		if (!route)
+			return -ENOMEM;
+
+		route->source = kstrdup(elem->source, GFP_KERNEL);
```

Comment:
Need to check return value here and deal with allocation failures.

Reply:
ok, I will do this.